src/librustc/mir/interpret/mod.rs

```diff
@@ -554,3 +556,70 @@ pub fn truncate(value: u128, size: Size) -> u128 {
     // Truncate (shift left to drop out leftover values, shift right to fill with zeroes).
     (value << shift) >> shift
 }
+
+
+impl<'tcx> TyCtxt<'tcx> {
```

Comment:
Why are these added in this file, as opposed to next to `const_eval` or so?

Scrolling over this file, the only coherent description of it that I can come up with is "random stuff".^^

Reply:
They need to be within `librustc` as they are used inside it. I'm not sure where the best place is.

Reply:
maybe just put them into a submodule of this module. `queries` could work as a name.